src/current/v23.2/create-procedure.md

```diff
@@ -0,0 +1,142 @@
+---
+title: CREATE PROCEDURE
+summary: The CREATE PROCEDURE statement creates a stored procedure.
+toc: true
+keywords:
+docs_area: reference.sql
+---
+
+The `CREATE PROCEDURE` [statement]({% link {{ page.version.version }}/sql-statements.md %}) defines a [stored procedure]({% link {{ page.version.version }}/stored-procedures.md %}).
+
+## Required privileges
+
+- To define a procedure, a user must have [`CREATE` privilege]({% link {{ page.version.version }}/security-reference/authorization.md %}#supported-privileges) on the schema of the procedure.
+- To define a procedure with a [user-defined type]({% link {{ page.version.version }}/create-type.md %}), a user must have [`USAGE` privilege]({% link {{ page.version.version }}/security-reference/authorization.md %}#supported-privileges) on the user-defined type.
+- To resolve a procedure, a user must have at least the [`USAGE` privilege]({% link {{ page.version.version }}/security-reference/authorization.md %}#supported-privileges) on the schema of the procedure.
+- To [call a procedure]({% link {{ page.version.version }}/call.md %}), a user must have [`EXECUTE` privilege]({% link {{ page.version.version }}/security-reference/authorization.md %}#supported-privileges) on the procedure.
+- At procedure definition and execution time, a user must have privileges on all the objects referenced in the procedure body. Privileges on referenced objects can be revoked and later procedure calls can fail due to lack of permission.
+
+If you grant `EXECUTE` privilege as a default privilege at the database level, newly created procedures inherit that privilege from the database.
+
+## Synopsis
+
+<div>
+{% remote_include https://raw.githubusercontent.com/cockroachdb/generated-diagrams/{{ page.release_info.crdb_branch_name }}/grammar_svg/create_proc.html %}
+</div>
+
+## Parameters
+
+|       Parameter       |                                                                   Description                                                                   |
+|-----------------------|-------------------------------------------------------------------------------------------------------------------------------------------------|
+| `routine_create_name` | The name of the procedure.                                                                                                                      |
+| `routine_param`       | A comma-separated list of procedure parameters.                                                                                                                          |
+| `routine_body_stmt`   | The body of the procedure. For allowed contents, see [Stored Procedures]({% link {{ page.version.version }}/stored-procedures.md %}#structure). |
```

Comment:
Should this include `routine_return_type`?

Reply:
I thought that procedures didn't return a value, and this isn't in the [generated SQL diagram](https://github.com/cockroachdb/cockroach/pull/117651). What does it refer to?

Reply:
Right, my mistake, just confused myself.